internal/scorecard/plugins/plugin_runner.go

```diff
@@ -178,59 +179,63 @@ func RunInternalPlugin(pluginType PluginType, config BasicAndOLMPluginConfig,
 	for _, cr := range config.CRManifest {
-		crSuites, err := runTests(csv, pluginType, config, cr, logFile)
+		crTestResults, _, err := runTests(csv, pluginType, config, cr, logFile)
 		if err != nil {
 			return scapiv1alpha2.ScorecardOutput{}, err
 		}
-		suites = append(suites, crSuites...)
+		testResults = append(testResults, crTestResults...)
 	}
 
-	output := schelpers.TestSuitesToScorecardOutput(suites, "")
-	return output, nil
+	output := scapiv1alpha2.NewScorecardOutput()
+	output.Log = ""
+
+	for _, tr := range testResults {
+		output.Results = append(output.Results, testResultToScorecardTestResult(tr))
+	}
+
+	return *output, nil
 }
 
 func ListInternalPlugin(pluginType PluginType, config BasicAndOLMPluginConfig) (scapiv1alpha2.ScorecardOutput, error) {
-	var suites []schelpers.TestSuite
+	testResults := make([]schelpers.TestResult, 0)
+	tests := make([]schelpers.Test, 0)
 
 	switch pluginType {
 	case BasicOperator:
 		conf := BasicTestConfig{}
-		basicTests := NewBasicTestSuite(conf)
-
-		basicTests.ApplySelector(config.Selector)
-
-		basicTests.TestResults = make([]schelpers.TestResult, 0)
-		for i := 0; i < len(basicTests.Tests); i++ {
-			result := schelpers.TestResult{}
-			result.State = scapiv1alpha2.PassState
-			result.Test = basicTests.Tests[i]
-			result.Suggestions = make([]string, 0)
-			result.Errors = make([]error, 0)
-			basicTests.TestResults = append(basicTests.TestResults, result)
-		}
-		suites = append(suites, *basicTests)
+		tests = append(tests, NewCheckSpecTest(conf))
+		tests = append(tests, NewCheckStatusTest(conf))
+		tests = append(tests, NewWritingIntoCRsHasEffectTest(conf))
 	case OLMIntegration:
 		conf := OLMTestConfig{}
-		olmTests := NewOLMTestSuite(conf)
-
-		olmTests.ApplySelector(config.Selector)
-
-		olmTests.TestResults = make([]schelpers.TestResult, 0)
-		for i := 0; i < len(olmTests.Tests); i++ {
-			result := schelpers.TestResult{}
-			result.State = scapiv1alpha2.PassState
-			result.Test = olmTests.Tests[i]
-			result.Suggestions = make([]string, 0)
-			result.Errors = make([]error, 0)
-			olmTests.TestResults = append(olmTests.TestResults, result)
-		}
-		suites = append(suites, *olmTests)
+		tests = append(tests, NewBundleValidationTest(conf))
+		tests = append(tests, NewCRDsHaveValidationTest(conf))
+		tests = append(tests, NewCRDsHaveResourcesTest(conf))
+		tests = append(tests, NewSpecDescriptorsTest(conf))
+		tests = append(tests, NewStatusDescriptorsTest(conf))
+	}
+
+	tests = applySelector(tests, config.Selector)
+
+	for i := 0; i < len(tests); i++ {
+		result := schelpers.TestResult{}
+		result.State = scapiv1alpha2.PassState
+		result.Test = tests[i]
+		result.Suggestions = make([]string, 0)
+		result.Errors = make([]error, 0)
+		testResults = append(testResults, result)
+	}
+
+	output := scapiv1alpha2.NewScorecardOutput()
+	output.Log = ""
+
+	for _, tr := range testResults {
+		output.Results = append(output.Results, testResultToScorecardTestResult(tr))
```

Comment:
Nit: It seems like we repeat this code in a few places. Would it make sense to pass the results and log values to `scapiv1alpha2.NewScorecardOutput`?

Reply:
I think I can fix that, but I'd like to move it to a separate PR if that's ok...in that PR I'd fix this, probably try to remove the helper package altogether, and restructure or combine the Run()/List() into a single chunk of code.